plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialect.java

```diff
@@ -441,7 +441,15 @@ default boolean supportsAliasInConditions() {
      */
     String formatStoredProcedureCall(DBPDataSource dataSource, String sqlText);
 
-    void generateStoredProcedureCall(StringBuilder sql, DBSProcedure proc, Collection<? extends DBSProcedureParameter> parameters);
+    /**
+     * Generates stored procedure call. Parameters (optionally) can be surrounded by cast(:param as paramType).
+     */
+    void generateStoredProcedureCall(
+        StringBuilder sql, 
+        DBSProcedure proc, 
+        Collection<? extends DBSProcedureParameter> parameters,
+        boolean castParams
+    );
```

Comment:
Please specify nullability annotations for the method's parameters. Don't forget about its implementations.

Reply:
Please,  give me an example. I just edited your file - so it (maybe) do not meet your new requirement?

Reply:
We use annotations to specify function arguments and return type nullability: @NotNull and @Nullable
It can be done like the following in this case
```
    void generateStoredProcedureCall(
        @NotNull StringBuilder sql, 
        @NotNull DBSProcedure proc, 
        @NotNull Collection<? extends DBSProcedureParameter> parameters,
        boolean castParams
    );
```

Reply:
I edited EXISTING method and i see that you dont used it while developing in. Suggest my further activities ? I can only say about parameter i added.
Why do you require code cleanup for the codebase you (or pecessors) was created?

Reply:
It's always nice to clean up an older code, especially when editing it anyways.

Reply:
> We use annotations to specify function arguments and return type nullability: @NotNull and @nullable It can be done like the following in this case
> 
> ```
>     void generateStoredProcedureCall(
>         @NotNull StringBuilder sql, 
>         @NotNull DBSProcedure proc, 
>         @NotNull Collection<? extends DBSProcedureParameter> parameters,
>         boolean castParams
>     );
> ```

Please confirm (fix) the method declaration and add (missing) code examples of checking for 'nullability'.  thank you

Reply:
@ShadelessFox I don't see the reason to strongly require it. Can we pass it to QA please?

Reply:
It's not a strict requirement.